lightning/src/ln/channelmanager.rs

```diff
@@ -284,6 +286,21 @@ struct PeerState {
 #[cfg(not(any(target_pointer_width = "32", target_pointer_width = "64")))]
 const ERR: () = "You need at least 32 bit pointers (well, usize, but we'll assume they're the same) for ChannelManager::latest_block_height";
 
+/// SimpleArcChannelManager is useful when you need a ChannelManager with a static lifetime, e.g.
+/// when you're using lightning-net-tokio (since tokio::spawn requires parameters with static
+/// lifetimes). Other times you can afford a reference, which is more efficient, in which case
+/// SimpleRefChannelManager is the more appropriate type. Defining these type aliases prevents
+/// issues such as overly long function definitions.
+pub type SimpleArcChannelManager<CS, M> = Arc<ChannelManager<CS, Arc<M>>>;
+
+/// SimpleRefChannelManager is a type alias for a ChannelManager reference, and is the reference
+/// counterpart to the SimpleArcChannelManager type alias. Use this type by default when you don't
+/// need a ChannelManager with a static lifetime. You'll need a static lifetime in cases such as
+/// usage of lightning-net-tokio (since tokio::spawn requires parameters with static lifetimes).
+/// But if this is not necessary, using a reference is more efficient. Defining these type aliases
+/// helps with issues such as long function definitions.
+pub type SimpleRefChannelManager<'a, 'b, CS, M> = &'a ChannelManager<CS, &'b M>;
```

Comment:
Same here, dont think it needs the outside reference. Also would be nice to use it in the tests.

Reply:
>Also would be nice to use it in the tests.

Hmm, the tests use `EnforcingChannelKeys` but I switched the SimpleRefChannelManager to use `InMemoryChannelKeys` per this comment: https://github.com/rust-bitcoin/rust-lightning/pull/443#discussion_r370439863 

Could create another `SimpleRefChannelManager` type that uses EnforcingChannelKeys instead!